src/sinol_make/commands/gen/__init__.py

```diff
@@ -0,0 +1,147 @@
+import argparse
+import glob
+import os
+import subprocess
+import hashlib
+import yaml
+
+import multiprocessing as mp
+
+from sinol_make import util
+from sinol_make.commands.gen import gen_util
+from sinol_make.commands.gen.structs import OutputGenerationArguments
+from sinol_make.helpers import parsers, package_util, compile
+from sinol_make.interfaces.BaseCommand import BaseCommand
+
+class Command(BaseCommand):
+    """
+    Class for `gen` command.
+    """
+
+    def get_name(self):
+        return "gen"
+
+    def configure_subparser(self, subparser):
+        parser = subparser.add_parser(
+            self.get_name(),
+            help='Generate input and output files',
+            description='Generate input files using gingen program '
+                        '(for example prog/abcingen.cpp for abc task). If there '
+                        'is a new test or it differs from the previous one, '
+                        'then the output file will be generated using the '
+                        'correct solution. You can also specify your ingen source '
+                        'file which will be used.'
+        )
+
+        parser.add_argument('ingen_path', type=str, nargs='?',
```

Comment:
What `nargs='?'` means?

Reply:
That this argument is optional and if present, only one string is taken